shotover-proxy/src/transforms/cassandra/sink_cluster/node_pool.rs

```diff
@@ -0,0 +1,86 @@
+use super::routing_key::calculate;
+use super::token_map::TokenMap;
+use crate::transforms::cassandra::sink_cluster::node::CassandraNode;
+use cassandra_protocol::frame::message_execute::BodyReqExecuteOwned;
+use cassandra_protocol::frame::message_result::PreparedMetadata;
+use cassandra_protocol::frame::Version;
+use cassandra_protocol::token::Murmur3Token;
+use cassandra_protocol::types::CBytesShort;
+use rand::prelude::*;
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub struct NodePool {
+    prepared_metadata: HashMap<CBytesShort, PreparedMetadata>,
+    token_map: TokenMap,
+    pub nodes: Vec<CassandraNode>,
+}
+
+impl NodePool {
+    pub fn new(nodes: Vec<CassandraNode>) -> Self {
+        Self {
+            token_map: TokenMap::new(nodes.as_slice()),
+            nodes,
+            prepared_metadata: HashMap::new(),
+        }
+    }
+
+    pub fn set_nodes(&mut self, nodes: Vec<CassandraNode>) {
+        self.nodes = nodes;
+        self.token_map = TokenMap::new(self.nodes.as_slice());
+    }
+
+    pub fn add_prepared_result(&mut self, id: CBytesShort, metadata: PreparedMetadata) {
+        self.prepared_metadata.insert(id, metadata);
+    }
+
+    pub fn random_node(&mut self, rng: &mut SmallRng) -> &mut CassandraNode {
```

Comment:
I dont think any of the usages of this function justify sending to a completely random node instead of to a random node within the rack we are proxying to.
Lets remove this function and call `get_random_node_in_dc_rack` instead.